cmd/influx/config/config.go

```diff
@@ -14,78 +14,144 @@ import (
 
 // Config store the crendentials of influxdb host and token.
 type Config struct {
+	Name string `toml:"-" json:"-"`
 	Host string `toml:"url" json:"url"`
 	// Token is base64 encoded sequence.
-	Token  string `toml:"token" json:"token"`
-	Org    string `toml:"org" json:"org"`
-	Active bool   `toml:"active" json:"active"`
+	Token          string `toml:"token" json:"token"`
+	Org            string `toml:"org" json:"org"`
+	Active         bool   `toml:"active,omitempty" json:"active,omitempty"`
+	PreviousActive bool   `toml:"previous,omitempty" json:"previous,omitempty"`
 }
 
 // DefaultConfig is default config without token
 var DefaultConfig = Config{
-	Host:   "http://localhost:9999",
-	Active: true,
+	Name: "default",
+	Host: "http://localhost:9999",
 }
 
 // Configs is map of configs indexed by name.
 type Configs map[string]Config
 
 // ConfigsService is the service to list and write configs.
 type ConfigsService interface {
-	WriteConfigs(pp Configs) error
-	ParseConfigs() (Configs, error)
+	CreateConfig(Config) (Config, error)
+	DeleteConfig(name string) (Config, error)
+	UpdateConfig(Config) (Config, error)
+	SwitchConfig(name string) (Config, error)
+	ListConfigs() (Configs, error)
+}
+
+// store is the embedded store of the Config service.
+type store interface {
+	parsePreviousActive() (Config, error)
+	ListConfigs() (Configs, error)
+	writeConfigs(cfgs Configs) error
 }
 
 // Switch to another config.
-func (pp *Configs) Switch(name string) error {
-	pc := *pp
+func (cfgs *Configs) Switch(name string) error {
```

Comment:
I just noticed this, `Configs` is a `map`, which means it is a reference type.  The referencing on the line below, doesn't do what I suspect it is intended to be protecting against. It'd be better to remove the `*` on all methods here and to stop dereferencing. You can still mutate the backing hashmap even if it is dereferenced